src/Dummy.cpp

```diff
@@ -9,27 +9,41 @@ const Dummy::uid_t Dummy::Join(const std::string &addr) {
   ++user_count;
   uid2addr.emplace(LastPlayer(), addr);
```

Comment:
what is addr?

Reply:
The user identity given from client or grpc adapter.

Reply:
grpc uses unsigned int to mark a user.

Reply:
There will be an adaptor to handle this. Binding with `string` provides extensibility.